keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -319,25 +349,33 @@ class TitusClusterHandler(
       .also { them ->
         if (them.distinctBy { it.container.digest }.size == 1) {
           val container = them.first().container
-          val images = cloudDriverService.findDockerImages(
-            account = getRegistryForTitusAccount(resource.spec.locations.account),
-            repository = container.repository()
-          ).filter { it.digest == container.digest }
-          if (images.isNotEmpty()) {
-            images.forEach { image ->
+          getTagsForDigest(container, resource.spec.locations.account)
+            .forEach { tag ->
               // We publish an event for each tag that matches the digest
               // so that we handle the tags like `latest` where more than one tags have the same digest
               // and we don't care about some of them.
               publisher.publishEvent(ArtifactVersionDeployed(
                 resourceId = resource.id,
-                artifactVersion = image.tag,
+                artifactVersion = tag,
                 provider = "titus"
               ))
             }
-          }
         }
       }
 
+  /**
+   * Get all tags that match a digest.
+   * Note: there may be more than one tag with the same digest (like `latest` and a version tag)
```

Comment:
I see that we end up filtering the `"latest"` tag at least for the `ArtifactVersionDeploying` event above. Should we filter it out here, at the "source"?

Reply:
hmm, ya it's probably better to do here since we'll never want `latest` in this class.